### PR TITLE
[JENKINS-65862] update Subversion to remove `asm` conflicts

### PR DESCRIPTION
**Pull request by car-roll**
Downstream of https://github.com/jenkinsci/subversion-plugin/pull/257

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
FYI, I opened jenkinsci/bom#550 to cover adding Subversion to the BOM's managed set. If accepted and released, and if `workflow-multibranch`, `workflow-cps-global-lib`, and `workflow-scm-step` adopted that release, then we would no longer need to specify a Subversion version in those plugins, which would simplify maintenance.

**Comment by basil** [comment]
Done in #52.

File: pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
         <changelist>-SNAPSHOT</changelist>
         <jenkins.version>2.222.4</jenkins.version>
         <java.level>8</java.level>
-        <subversion-plugin.version>2.14.0</subversion-plugin.version>
+        <subversion-plugin.version>2.14.3-rc1161.5aa3d92d94e9</subversion-plugin.version> <!-- TODO: https://github.com/jenkinsci/subversion-plugin/pull/257 -->
```
Comment (alecharp) [inline pom.xml:45]: The PR was merged and released so

```suggestion
        <subversion-plugin.version>2.14.3</subversion-plugin.version>
```

**Review by alecharp — CHANGES_REQUESTED** [review]

**Review by MRamonLeon — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (car-roll) [reply pom.xml:45]: Going to bump subversion to 2.14.4 to take in https://github.com/jenkinsci/subversion-plugin/pull/259 where digester is removed completely. See https://github.com/jenkinsci/subversion-plugin/pull/258 for context

Comment (car-roll) [reply pom.xml:45]: Closing in favor of https://github.com/jenkinsci/workflow-scm-step-plugin/pull/52

**Comment by car-roll** [comment]
Closing in favor of https://github.com/jenkinsci/workflow-scm-step-plugin/pull/52

**Pull request closed without merging**